lib/plugins/aws/deploy/compile/events/apiGateway/lib/validate.js

```diff
@@ -202,7 +202,7 @@ module.exports = {
     } else if (typeof authorizer === 'object') {
       if (authorizer.arn) {
         arn = authorizer.arn;
-        name = this.provider.naming.extractAuthorizerNameFromArn(arn);
+        name = typeof authorizer.name == 'string' ? authorizer.name : this.provider.naming.extractAuthorizerNameFromArn(arn);
```

Comment:
There should also be fixes in `extractAuthorizerNameFromArn()` as the malicious name generation happens there in the first place